ProcessLib/Process.cpp

```diff
@@ -50,6 +50,38 @@ Process::Process(
 {
 }
 
+/// Initialize the boundary conditions for single PDE
+void Process::initializeBoundaryConditionPerPDE(
+    const NumLib::LocalToGlobalIndexMap& dof_table, const int process_id)
+{
+    auto const& per_process_variables = _process_variables[process_id];
+    auto& per_process_BCs = _boundary_conditions[process_id];
+
+    per_process_BCs.addBCsForProcessVariables(per_process_variables, dof_table,
+                                              _integration_order);
+
+    std::vector<std::unique_ptr<NodalSourceTerm>> per_process_source_terms;
+    for (std::size_t variable_id = 0;
+         variable_id < per_process_variables.size();
+         variable_id++)
+    {
+        ProcessVariable& pv = per_process_variables[variable_id];
+        auto sts = pv.createSourceTerms(dof_table, 0, _integration_order);
+
+        std::move(sts.begin(), sts.end(),
+                  std::back_inserter(per_process_source_terms));
+    }
+    _source_terms.push_back(std::move(per_process_source_terms));
+}
+
+void Process::initializeBoundaryConditions()
+{
+    for (std::size_t pcs_id = 0; pcs_id < _process_variables.size(); pcs_id++)
```

Comment:
Mismatch: counting of **process ids** but using **process variable**s' size. I guess it is correct, but the names are misleading.

Reply:
Added a description above the loop.